lib/exhal/collection.ex

```diff
@@ -7,14 +7,28 @@ defmodule ExHal.Collection do
   alias ExHal.ResponseHeader
 
   @doc """
-  Returns a stream that iterate over the collection represented by `a_doc`.
+  Returns a stream that iterates over the collection represented by `a_doc`.
+  Iteration halts when there is no further `next` link to follow.
+
+  If an `ExHal.Error` occurs when requesting the next resource, an
+  `ExHal.CollectionError` will be raised to the caller.
```

Comment:
Maybe be a little clearer that a collection error will be raised only when following the next link to the the next page? This currently reads a little like it might raise a collection error if one of the members of the collection is unfetchable.

Reply:
updated

Reply:
updated